graylog2-web-interface/src/views/components/QueryBar.tsx

```diff
@@ -68,13 +65,21 @@ type Props = {
 
 const QueryBar = ({ queries, queryTitles, viewMetadata }: Props) => {
   const { activeQuery } = viewMetadata;
-  const onRemove = useCallback((queryId) => onCloseTab(queryId, activeQuery, queries), [activeQuery, queries]);
+  const { setDashboardPage } = useContext(DashboardPageContext);
+
+  const onSelectPage = useCallback((pageId) => {
+    setDashboardPage(pageId);
+
+    return onSelectQuery(pageId);
+  }, [setDashboardPage]);
```

Comment:
An alternative solution could be, removing `onSelectQuery` and changing  `onSelectPage ` to:

```
  const onSelectPage = useCallback((pageId) => {
    if (pageId === 'new') {
      return NewQueryActionHandler().then((newPage) => {
        setDashboardPage(newPage.id);
      });
    }

    setDashboardPage(pageId);

    return ViewActions.selectQuery(pageId);
  }, [setDashboardPage]);
```

This way we are syncing the URL after creating a new page. Currently we are not defining the `page` URL query param after creating a new page. Another benefit, the `DashboardPageContext` will be slightly less complex, because we do not have to consider that the page id can be `new`.